// @syncfusion/ej2-angular-gantt/esm2020/src/gantt/gantt.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ColumnDirective, ColumnsDirective } from './columns.directive';
import { AddDialogFieldDirective, AddDialogFieldsDirective } from './adddialogfields.directive';
import { EditDialogFieldDirective, EditDialogFieldsDirective } from './editdialogfields.directive';
import { DayWorkingTimeDirective, DayWorkingTimeCollectionDirective } from './dayworkingtime.directive';
import { HolidayDirective, HolidaysDirective } from './holidays.directive';
import { EventMarkerDirective, EventMarkersDirective } from './eventmarkers.directive';
import { GanttComponent } from './gantt.component';
import * as i0 from "@angular/core";
/**
 * NgModule definition for the Gantt component.
 */
export class GanttModule {
}
GanttModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: GanttModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
GanttModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: GanttModule, declarations: [GanttComponent,
        ColumnDirective,
        ColumnsDirective,
        AddDialogFieldDirective,
        AddDialogFieldsDirective,
        EditDialogFieldDirective,
        EditDialogFieldsDirective,
        DayWorkingTimeDirective,
        DayWorkingTimeCollectionDirective,
        HolidayDirective,
        HolidaysDirective,
        EventMarkerDirective,
        EventMarkersDirective], imports: [CommonModule], exports: [GanttComponent,
        ColumnDirective,
        ColumnsDirective,
        AddDialogFieldDirective,
        AddDialogFieldsDirective,
        EditDialogFieldDirective,
        EditDialogFieldsDirective,
        DayWorkingTimeDirective,
        DayWorkingTimeCollectionDirective,
        HolidayDirective,
        HolidaysDirective,
        EventMarkerDirective,
        EventMarkersDirective] });
GanttModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: GanttModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: GanttModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        GanttComponent,
                        ColumnDirective,
                        ColumnsDirective,
                        AddDialogFieldDirective,
                        AddDialogFieldsDirective,
                        EditDialogFieldDirective,
                        EditDialogFieldsDirective,
                        DayWorkingTimeDirective,
                        DayWorkingTimeCollectionDirective,
                        HolidayDirective,
                        HolidaysDirective,
                        EventMarkerDirective,
                        EventMarkersDirective
                    ],
                    exports: [
                        GanttComponent,
                        ColumnDirective,
                        ColumnsDirective,
                        AddDialogFieldDirective,
                        AddDialogFieldsDirective,
                        EditDialogFieldDirective,
                        EditDialogFieldsDirective,
                        DayWorkingTimeDirective,
                        DayWorkingTimeCollectionDirective,
                        HolidayDirective,
                        HolidaysDirective,
                        EventMarkerDirective,
                        EventMarkersDirective
                    ]
                }]
        }] });
